krita_batch_exporter/Infrastructure.py

```diff
@@ -19,13 +19,48 @@ def nodeToImage(wnode):
     """
     Returns an QImage 8-bit sRGB
     """
+
+
+    SRGB_PROFILE = "sRGB-elle-V2-srgbtrc.icc"
+
     [x, y, w, h] = wnode.bounds
-    return QImage(wnode.node.thumbnail(w, h))
 
 
-def expand_and_format(img, margin=0, is_jpg=False):
+    is_srgb = (wnode.node.colorModel() == "RGBA"
+        and wnode.node.colorDepth() == "U8"
+        and wnode.node.colorProfile().lower() == SRGB_PROFILE.lower())
+
+    if is_srgb:
+        pixel_data = wnode.node.projectionPixelData(x, y, w, h).data()
+    else:
+        temp_doc = KI.createDocument(
+            w,
+            h,
+            "_batch_exporter_temp",
+            wnode.node.colorModel(),
+            wnode.node.colorDepth(),
+            wnode.node.colorProfile(),
+            72
+        )
+
+        temp_layer = temp_doc.topLevelNodes()[0]
+        pixel_data = wnode.node.projectionPixelData(x, y, w, h).data()
+        temp_layer.setPixelData(pixel_data, x, y, w, h)
+
+        temp_doc.setColorSpace("RGBA", "U8", SRGB_PROFILE)
+        assert temp_doc.colorModel() == "RGBA" and temp_doc.colorDepth() == "U8"
+        assert temp_layer.colorModel() == "RGBA" and temp_layer.colorDepth() == "U8"
+
+        pixel_data = temp_layer.projectionPixelData(x, y, w, h).data()
+
+        temp_doc.close()
```

Comment:
Seems like all of this stuff can be replaced with:
```suggestion
        temp_node = wnode.node.duplicate()
        temp_node.setColorSpace(wnode.node.colorModel(), wnode.node.colorDepth(), SRGB_PROFILE)
        assert temp_node.colorModel() == "RGBA" and temp_node.colorDepth() == "U8"
        pixel_data = temp_node.projectionPixelData(x, y, w, h).data()
        del temp_node
```

Can you please confirm at your end that it doesn't break anything? If all's good looks like it's ready to merge.

Reply:
Sure, I can test it today after working hours.